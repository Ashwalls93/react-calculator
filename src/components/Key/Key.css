.key-container {
  align-items: center;
  display: flex;
  height: 25%;
  justify-content: center;
  transition: background-color 0.3s linear;
}

.key-container:hover {
  cursor: pointer;
}

.operator-key {
  width: 100%;
}

.operator-key-theme-default {
  background-color: var(--operator-key-color-default);
}

.operator-key-theme-default:hover {
  background-color: var(--operator-key-color-default-hover); 
}

.operator-key-theme-pony {
  background-color: var(--operator-key-color-pony);
}

.operator-key-theme-pony:hover {
  background-color: var(--operator-key-color-pony-hover); 
}

.operator-key-theme-dondi {
  background-color: var(--operator-key-color-dondi);
  box-shadow: var(--key-border);
}

.operator-key-theme-dondi:hover {
  background-color: var(--operator-key-color-dondi-hover); 
}

.number-key {
  width: calc(100%/3);
}

.number-key-theme-default {
  background-color: var(--number-key-color-default);
}

.number-key-theme-default:hover {
  background-color: var(--number-key-color-default-hover);
}

.number-key-theme-pony {
  background-color: var(--number-key-color-pony);
  border-radius: 8px;
}

.number-key-theme-pony:hover {
  background-color: var(--number-key-color-pony-hover);
}

.number-key-theme-dondi {
  background-color: var(--number-key-color-dondi);
  border-radius: 8px;
  box-shadow: var(--key-border);
}

.number-key-theme-dondi:hover {
  background-color: var(--number-key-color-dondi-hover);
}

.submit-key {
  height: 100%;
  width: 75%;
}

.submit-key-theme-default {
  background-color: var(--submit-key-color-default);
}

.submit-key-theme-default:hover {
  background-color: var(--submit-key-color-default-hover);
}

.submit-key-theme-pony {
  border-radius: 8px;
  background-color: var(--submit-key-color-pony);
}

.submit-key-theme-pony:hover {
  background-color: var(--submit-key-color-pony-hover);
}

.submit-key-theme-dondi {
  background-color: var(--submit-key-color-dondi);
  border-radius: 8px;
  box-shadow: var(--key-border);
}

.submit-key-theme-dondi:hover {
  background-color: var(--submit-key-color-dondi-hover);
}

.first-operator {
  border-radius: 8px 8px 0 0;
}

.last-operator {
  border-radius: 0 0 8px 8px;
} 

.key-value {
  color: var(--key-text-color);
  font-family: var(--main-font);
  font-size: var(--key-text-size);
  font-weight: var(--key-text-weight);
}

.key-theme-default {
  color: var(--key-text-color-default);
}

.key-theme-pony {
  color: var(--key-text-color-pony);
}

.key-theme-dondi {
  color: var(--key-text-color-dondi);
}