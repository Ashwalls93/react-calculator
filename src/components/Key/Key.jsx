import React from 'react';
import PropTypes from 'prop-types';
import './Key.css';

const Key = ({theme, keyAction, keyType, keyValue}) => {
  let keyClassName = `key-container ${keyType} ${keyType}-theme-${theme}`
  
  if (keyValue === '/' && theme !== 'default') keyClassName = `${keyClassName} first-operator`;
  if (keyValue === '+' && theme !== 'default') keyClassName = `${keyClassName} last-operator`;
  
  return (
    <div 
      className={keyClassName}
      onClick={() => {keyAction(keyValue)}}
    >
      <p className={`key-value key-theme-${theme}`}>
        {keyValue}
      </p>
    </div>
  );
}

Key.propTypes = {
  keyAction: PropTypes.func.isRequired,
  keyType: PropTypes.string.isRequired,
  keyValue: PropTypes.string.isRequired,
  theme: PropTypes.string.isRequired,
}

export default Key;