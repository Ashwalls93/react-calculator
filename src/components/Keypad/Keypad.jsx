import React from 'react';
import PropTypes from 'prop-types';
import Key from '../Key/Key';
import StarKey from '../StarKey/StarKey';
import './Keypad.css';

const Keypad = ({callOperator, toggleTheme, theme, numbers, operators, setOperator, updateDisplay}) => {

  numbers = numbers.map(number => {
    return (
      <Key
        key={number}
        keyAction={updateDisplay}
        keyType="number-key"
        keyValue={number}
        theme={theme}
      />
    );
  });

  operators = operators.map(operator => {
    return (
      <Key
        key={operator}
        keyAction={setOperator}
        keyType="operator-key"
        keyValue={operator}
        theme={theme}
      />
    );
  });

  return (
    <div className="keypad-container">
      <div className="numbers-container">
        {numbers}
      </div>
      <div className="operators-container">
        {operators}
      </div>
      <div className="submit-container">
        <Key
          keyAction={callOperator}
          keyType="submit-key"
          keyValue="="
          theme={theme}
        />
        <StarKey 
          theme={theme}
          toggleTheme={toggleTheme}
        />
      </div>
    </div>
  );
}

Keypad.propTypes = {
  callOperator: PropTypes.func.isRequired,
  numbers: PropTypes.array.isRequired,
  operators: PropTypes.array.isRequired,
  setOperator: PropTypes.func.isRequired,
  theme: PropTypes.string.isRequired,
  toggleTheme: PropTypes.func.isRequired,
  updateDisplay: PropTypes.func.isRequired,
}

export default Keypad;