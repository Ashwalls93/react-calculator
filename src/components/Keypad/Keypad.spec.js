import React from 'react';
import {mount, shallow} from 'enzyme';
import Keypad from './Keypad';
import Key from '../Key/Key';

describe('Keypad', () => {
  it('should render 4 <div>\'s', () => {
    const wrapper = shallow(<Keypad numbers={[]} operators={[]} />);
    expect(wrapper.find('div').length).toEqual(4);
  });

  it('renders the values of numbers to the DOM', () => {
    const wrapper = mount(<Keypad numbers={[]} operators={[]} />);
    wrapper.setProps({numbers: ['0', '1', '2']});
    expect(wrapper.text()).toEqual('012=');
  });

  it('renders the values of operators to the DOM', () => {
    const wrapper = mount(<Keypad numbers={[]} operators={[]} />);
    wrapper.setProps({operators: ['+', '-', '*', '/']});
    expect(wrapper.text()).toEqual('+-*/=');
  });

  it('should render the Key component', () => {
    const wrapper = mount(<Keypad numbers={[]} operators={[]} />);
    expect(wrapper.containsMatchingElement([
      <Key keyType="" keyValue="" />
    ])).toEqual(true);
  });

  it('should render the Key component for each index of numbers, operators, and the submit Key', () => {
    const numbers = ['0', '1'];
    const operators = ['+', '-'];
    const submit = 1;
    const keyTotal = numbers.length + operators.length + submit;
    const wrapper = shallow(<Keypad numbers={numbers} operators={operators} />);
    expect(wrapper.find('Key').length).toEqual(keyTotal);
  });
});