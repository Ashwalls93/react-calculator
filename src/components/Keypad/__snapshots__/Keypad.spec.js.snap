// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keypad should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Keypad
    callOperator={[Function]}
    numbers={Array []}
    operators={Array []}
    setOperator={[Function]}
    theme="default"
    toggleTheme={[Function]}
    updateDisplay={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="numbers-container"
>
          
</div>,
        <div
          className="operators-container"
>
          
</div>,
        <div
          className="submit-container"
>
          <Key
                    keyAction={[Function]}
                    keyType="submit-key"
                    keyValue="="
                    theme="default"
          />
          <StarKey
                    theme="default"
                    toggleTheme={[Function]}
          />
</div>,
      ],
      "className": "keypad-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "numbers-container",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "operators-container",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Key
              keyAction={[Function]}
              keyType="submit-key"
              keyValue="="
              theme="default"
/>,
            <StarKey
              theme="default"
              toggleTheme={[Function]}
/>,
          ],
          "className": "submit-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "keyAction": [Function],
              "keyType": "submit-key",
              "keyValue": "=",
              "theme": "default",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "theme": "default",
              "toggleTheme": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="numbers-container"
>
            
</div>,
          <div
            className="operators-container"
>
            
</div>,
          <div
            className="submit-container"
>
            <Key
                        keyAction={[Function]}
                        keyType="submit-key"
                        keyValue="="
                        theme="default"
            />
            <StarKey
                        theme="default"
                        toggleTheme={[Function]}
            />
</div>,
        ],
        "className": "keypad-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "numbers-container",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "operators-container",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Key
                keyAction={[Function]}
                keyType="submit-key"
                keyValue="="
                theme="default"
/>,
              <StarKey
                theme="default"
                toggleTheme={[Function]}
/>,
            ],
            "className": "submit-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "keyAction": [Function],
                "keyType": "submit-key",
                "keyValue": "=",
                "theme": "default",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "theme": "default",
                "toggleTheme": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
